packages/flutter_driver/lib/src/driver/vmservice_driver.dart

```diff
@@ -96,11 +96,22 @@ class VMServiceFlutterDriver extends FlutterDriver {
     final VMServiceClientConnection connection =
     await vmServiceConnectFunction(dartVmServiceUrl, headers: headers);
     final VMServiceClient client = connection.client;
-    final VM vm = await client.getVM();
-    final VMIsolateRef isolateRef = isolateNumber ==
-        null ? vm.isolates.first :
-    vm.isolates.firstWhere(
-            (VMIsolateRef isolate) => isolate.number == isolateNumber);
+
+    VMIsolateRef isolateRef;
+    for (int tries = 0; tries < 10; tries += 1) {
+      final VM vm = await client.getVM();
+      if (vm.isolates.isEmpty) {
+        await Future<void>.delayed(_kPauseBetweenReconnectAttempts);
+        continue;
+      }
+      isolateRef = isolateNumber == null
+        ? vm.isolates.first
+        : vm.isolates.firstWhere((VMIsolateRef isolate) => isolate.number == isolateNumber);
+      break;
+    }
+    if (isolateRef == null) {
+      throw DriverException('Failed to find main isolate after ${_kPauseBetweenReconnectAttempts.inSeconds * 10} seconds!');
```

Comment:
Check `inSeconds * 10` reported as seconds.

Reply:
Because we looped 10 times

Reply:
Ah, ok. How about a const int instead of copying 10 in both places?

Reply:
done

Reply:
Ahh internal testing on my other patch is saying this might not be enough in the case where the isolate number is supplied. Will update.

Reply:
Nevermind. It's something wacky about the different patch.